### PR TITLE
eclipse/rdf4j#1197 add statements with correct graph

### PR DESCRIPTION
**Pull request by hmottestad**
This PR addresses GitHub issue: eclipse/rdf4j#1197 .

Briefly describe the changes proposed in this PR:

* inferred statements get graph based on added statement
* 
*

### CODE REVIEW

**Comment by hmottestad** [comment]
Need to merge the testfixes first: https://github.com/eclipse/rdf4j-testsuite/pull/40

**Comment by hmottestad** [comment]
@barthanssens @jeenbroekstra 

Had to combine multiple RDF files into one by having each file in a graph. Then used the ContextAwareRepository to read/write a single graph. Essentially getting a virtual repository that looked like everything was in the default graph, but instead it was just projecting a graph from the underlying sail which contained one graph per file. 

Found out that when doing this reasoning wasn't working as expected. So ended up having the RDFS reasoner adding the inferred statements to the context (graph) based on the context of the added statement. 

Do you guys think this makes sense to do?

**Comment by hmottestad** [comment]
@jeenbroekstra I've added a ```sail.setAddInferredStatementsToDefaultContext(boolean)``` method to the sail and set it to true as default.

If I remember correctly merging this to master makes it a minor release and then I can make a new PR with default as false and merge that into develop which will result in a major release? Or was it the other way around?

**Comment by jeenbroekstra** [comment]
@hmottestad merging to master makes it part of the next _patch_ release, merging to develop is for the next _minor_ release. We currently have no branch specifically for the next major release. 

So I think this needs to be merged to develop. As for a followup PR to make a change that needs a major release: it may be time for us to start a branch for development of a 3.0 release. Not to mention start planning around when we want to release this, and what should be part of it.

**Comment by jeenbroekstra** [comment]
No - the definition of a minor release is that it contains only backward compatible/non-breaking changes :)

File: inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/SchemaCachingRDFSInferencer.java [hunk 1]
```diff
@@ -93,6 +93,13 @@
 	// The previous transaction rolled back
 	boolean rolledBackAfterModifyingSchemaCache;
 
+	// Inferred statements can either be added to the default context
+	// or to the context that the original inserted statement has
+	// for the time being, the default behaviour will be to adde the
+	// statements to the default context.
+	// THIS BEHAVIOUR WILL BE SWITCHED ON THE NEXT MAJOR RELEASE
```
Comment (jeenbroekstra) [inline inferencer/src/main/java/org/eclipse/rdf4j/sail/inferencer/fc/SchemaCachingRDFSInferencer.java]: Can you log a new issue for this so we don't forget to pick this up for the 3.0 milestone?

**Pull request merged**